presto-native-execution/presto_cpp/main/types/TypeSignatureTypeConverter.cpp

```diff
@@ -121,6 +121,22 @@ antlrcpp::Any TypeSignatureTypeConverter::visitArray_type(
   return arrayFromType(visit(ctx->type()).as<TypePtr>());
 }
 
+antlrcpp::Any TypeSignatureTypeConverter::visitFunction_type(
+    TypeSignatureParser::Function_typeContext* ctx) {
+  const auto numArgs = ctx->type().size() - 1;
+
+  std::vector<TypePtr> argumentTypes;
+  argumentTypes.reserve(numArgs);
+  for (auto i = 0; i < numArgs; ++i) {
+    argumentTypes.push_back(visit(ctx->type()[i]).as<TypePtr>());
+  }
+
+  auto returnType = visit(ctx->type().back()).as<TypePtr>();
+
+  TypePtr functionType = FUNCTION(std::move(argumentTypes), returnType);
```

Comment:
```
nit: return FUNCTION(std::move(argumentTypes), returnType);?
```

Reply:
There are some details about how antlrcpp::Any works that I don't fully understand, but the suggested rewrite doesn't work. I spend ~ an hour debugging this earlier today.